helper.php

```diff
@@ -356,32 +373,31 @@ public function tpl_tags($print = true) {
     }
 
     /**
+     * @param string $namespace empty for entire wiki
+     * 
      * @return array
      */
-    public function getAllTags() {
-
+    public function getAllTags($namespace='', $order_by='tag', $desc=false) {
+        $order_fields = array('pid', 'tid', 'orig', 'taggers', 'count');
+        if (!in_array($order_by, $order_fields))
+            throw new Exception('cannot sort by '.$order_by. ' field does not exists');
+        
         $db = $this->getDb();
-        $res = $db->query('SELECT pid, tag, tagger FROM taggings ORDER BY tag');
-
-        $tags_tmp = $db->res2arr($res);
-        $tags = array();
-        foreach ($tags_tmp as $tag) {
-            $tid = $this->cleanTag($tag['tag']);
-
-            if (!isset($tags[$tid]['orig'])) {
-                $tags[$tid]['orig'] = array();
-            }
-            $tags[$tid]['orig'][] = $tag['tag'];
-
-            if (isset($tags[$tid]['count'])) {
-                $tags[$tid]['count']++;
-                $tags[$tid]['tagger'][] = $tag['tagger'];
-            } else {
-                $tags[$tid]['count'] = 1;
-                $tags[$tid]['tagger'] = array($tag['tagger']);
-            }
-        }
-        return $tags;
+        
+        $query = 'SELECT    pid,
+                            CLEANTAG(tag) as tid,
+                            GROUP_CONCAT(tag, ", ") AS orig,
+                            GROUP_CONCAT(tagger, ", ") AS taggers,
+                            COUNT(*) AS "count"
+                        FROM (SELECT * FROM "taggings" ORDER BY tagger) /*sort taggers inside GROUP_CONCAT*/
```

Comment:
I wonder if this subselect is a good idea. It probably makes it impossible for the outer select to use any indexes.

Reply:
I can use php sort instead.